### DISCUSSION TITLE
Marblenet vad for real time streaming applications

### DISCUSSION THREAD

**Discussion by arielrado** [NVIDIA-NeMo/NeMo#8326] [Q&A]
Hello,

I have been trying to use nvidia’s marblenet for voice activity detection for real time audio and have run into some trouble.

following the notebook from nemo’s [github](https://github.com/NVIDIA/NeMo/blob/main/tutorials/asr/Online_Offline_Microphone_VAD_Demo.ipynb), specifically the part talking about online microphone inference. When testing with some of my data I get inconsistent results. The probabilities of speech and non speech are very close to each other, reaching a verdict by a very thin margin (around 0.01), icreasing the threshold to anything above 0.5 results in constant non-speech labels.

Any insights are welcome!

**Comment by smallsudarshan** [comment]
I am facing a similar issue. My input is PCM int 16, 16khz audio and my application has a minimum step length of 0.02. Somehow, the probability is always around 0.57 regardless of whether there is speech or not. I have tried with larger step lengths but nothing changed.
